apiserver/root.go

```diff
@@ -918,12 +919,13 @@ func (ctx *facadeContext) HTTPClient(purpose facade.HTTPClientPurpose) (facade.H
 	return client, nil
 }
 
-var storageRegistryGetter = func(ctx *facadeContext) func() (storage.ProviderRegistry, error) {
-	return func() (storage.ProviderRegistry, error) {
+func storageRegistryGetter(ctx *facadeContext) corestorage.ModelStorageRegistryGetter {
```

Comment:
How do you feel if I expose the storage registry from the domain services, so that we can remove all the code from the apiserver?

It could be on the stub services for now?

Reply:
See: https://github.com/juju/juju/pull/18292 as a follow on.